mbf_utility/src/navigation_utility.cpp

```diff
@@ -211,65 +211,122 @@ double angle(const geometry_msgs::PoseStamped &pose1, const geometry_msgs::PoseS
+    return "Patience exceeded";
+  if (outcome == mbf_msgs::ExePathResult::COLLISION)
+    return "Collision";
+  if (outcome == mbf_msgs::ExePathResult::OSCILLATION)
+    return "Oscillation";
+  if (outcome == mbf_msgs::ExePathResult::ROBOT_STUCK)
+    return "Robot stuck";
+  if (outcome == mbf_msgs::ExePathResult::MISSED_GOAL)
+    return "Missed Goal";
+  if (outcome == mbf_msgs::ExePathResult::MISSED_PATH)
+    return "Missed Path";
+  if (outcome == mbf_msgs::ExePathResult::BLOCKED_GOAL)
+    return "Blocked Goal";
+  if (outcome == mbf_msgs::ExePathResult::BLOCKED_PATH)
+    return "Blocked Path";
+  if (outcome == mbf_msgs::ExePathResult::INVALID_PATH)
+    return "Invalid Path";
+  if (outcome == mbf_msgs::ExePathResult::TF_ERROR)
+    return "TF Error";
+  if (outcome == mbf_msgs::ExePathResult::NOT_INITIALIZED)
+    return "Not initialized";
+  if (outcome == mbf_msgs::ExePathResult::INVALID_PLUGIN)
+    return "Invalid Plugin";
+  if (outcome == mbf_msgs::ExePathResult::INTERNAL_ERROR)
+    return "Internal Error";
+  if (outcome == mbf_msgs::ExePathResult::OUT_OF_MAP)
+    return "Out of map";
+  if (outcome == mbf_msgs::ExePathResult::MAP_ERROR)
+    return "Map error";
+  if (outcome == mbf_msgs::ExePathResult::STOPPED)
+    return "Stopped";
+  if (outcome >= mbf_msgs::ExePathResult::PLUGIN_ERROR_RANGE_START &&
+      outcome <= mbf_msgs::ExePathResult::PLUGIN_ERROR_RANGE_END)
+    return "Plugin-specific controller error";
 
-    case mbf_msgs::RecoveryResult::FAILURE:         return "Failure";
-    case mbf_msgs::RecoveryResult::CANCELED:        return "Canceled";
-    case mbf_msgs::RecoveryResult::PAT_EXCEEDED:    return "Patience exceeded";
-    case mbf_msgs::RecoveryResult::TF_ERROR:        return "TF Error";
-    case mbf_msgs::RecoveryResult::NOT_INITIALIZED: return "Not initialized";
-    case mbf_msgs::RecoveryResult::INVALID_PLUGIN:  return "Invalid Plugin";
-    case mbf_msgs::RecoveryResult::INTERNAL_ERROR:  return "Internal Error";
-    case mbf_msgs::RecoveryResult::IMPASSABLE:      return "Impassable";
-    case mbf_msgs::RecoveryResult::STOPPED:         return "Stopped";
+  if (outcome == mbf_msgs::RecoveryResult::FAILURE)
+    return "Failure";
+  if (outcome == mbf_msgs::RecoveryResult::CANCELED)
+    return "Canceled";
+  if (outcome == mbf_msgs::RecoveryResult::PAT_EXCEEDED)
+    return "Patience exceeded";
+  if (outcome == mbf_msgs::RecoveryResult::TF_ERROR)
+    return "TF Error";
+  if (outcome == mbf_msgs::RecoveryResult::NOT_INITIALIZED)
+    return "Not initialized";
+  if (outcome == mbf_msgs::RecoveryResult::INVALID_PLUGIN)
+    return "Invalid Plugin";
+  if (outcome == mbf_msgs::RecoveryResult::INTERNAL_ERROR)
+    return "Internal Error";
+  if (outcome == mbf_msgs::RecoveryResult::IMPASSABLE)
+    return "Impassable";
+  if (outcome == mbf_msgs::RecoveryResult::STOPPED)
+    return "Stopped";
+  if (outcome >= mbf_msgs::RecoveryResult::PLUGIN_ERROR_RANGE_START &&
```

Comment:
S: keep the switch case, add this if at the start.
or, if you are ok of printing the enum as string (e.g. when you get `mbf_msgs::RecoveryResult::FAILURE` the output would be `mbf_msgs::RecoveryResult::FAILURE` as string), you can use magic_enum:

https://github.com/rapyuta-robotics/rr_amr_behaviors/blob/devel/rr_behavior_utils/src/utils.cpp#L233

Reply:
I wanted to keep codes grouped by action, that's why I removed the switch